processing/src/main/java/org/apache/druid/query/NoopQueryProcessingPool.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query;
+
+import org.apache.druid.java.util.common.concurrent.Execs;
+
+/**
+ * Implementation of {@link QueryProcessingPool} that throws when it is given any query execution task unit
+ */
+public class NoopQueryProcessingPool extends ForwardingQueryProcessingPool
```

Comment:
If we are writing a `Noop` implementation, it should not extend the `Forwarding` pool, rather implement the `QueryProcessingPool` directly and throw unsupported or equivalent exception in all methods.